### ISSUE TITLE
ValueError: invalid literal for int() with base 10

### ISSUE THREAD

**Issue by prasadkris** [sentry-kubernetes/charts#714]
Greetings, I am trying to deploy sentry on our self hosted K8s cluster (v1.22) using `sentry/sentry` chart `v15.1.0` with external clickhouse settings as shown below:

```
externalClickhouse:
  clusterName: clickhouse
  database: default
  host: clickhouse
  httpPort: 8123
  password: ""
  singleNode: false
  tcpPort: 9000
  username: default
```

The deployment fails and most of the pods stay in a `CrashLoopBackOff` state due to the below errors: 

```
kubectl -n sentry logs sentry-snuba-db-init--1-t67r7 
Traceback (most recent call last):
  File "/usr/local/bin/snuba", line 33, in <module>
    sys.exit(load_entry_point('snuba', 'console_scripts', 'snuba')())
  File "/usr/local/bin/snuba", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/local/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/src/snuba/snuba/cli/__init__.py", line 21, in <module>
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/src/snuba/snuba/cli/admin.py", line 6, in <module>
    from snuba.environment import setup_logging
  File "/usr/src/snuba/snuba/environment.py", line 13, in <module>
    from snuba import settings
  File "/usr/src/snuba/snuba/settings/__init__.py", line 33, in <module>
    "port": int(os.environ.get("CLICKHOUSE_PORT", 9000)),
ValueError: invalid literal for int() with base 10: 'tcp://10.103.178.193:9000'
```

Appreciate if someone could help me with this, thanks!

**Comment by junowong0114** [comment]
Your problem seems to be on this line in snuba source code: https://github.com/getsentry/snuba/blob/22.7.0/snuba/settings/__init__.py#L33

Try `helm upgrade --dry-run`. Would you double check the generated yaml file of your snuba configmap? Particularly, what is the value of `CLUSTERS[0].port`?

**Comment by prasadkris** [comment]
Thank you, @junowong0114 🙏  The `CLUSTERS[0].port` is set to `int(9000),` as seen below, I tried a deployment after updating it to  `"port": 9000` but it didn't help :-(

```
   # Clickhouse Options
    CLUSTERS = [
      {
        "host": env("CLICKHOUSE_HOST", "clickhouse"),
        "port": int(9000),
        "user":  env("CLICKHOUSE_USER", "default"),
        "password": env("CLICKHOUSE_PASSWORD", ""),
        "database": env("CLICKHOUSE_DATABASE", "default"),
        "http_port": 8123,
```

**Comment by junowong0114** [comment]
@prasadkris Judging from these lines:
```
File "/usr/src/snuba/snuba/settings/__init__.py", line 33, in <module>
    "port": int(os.environ.get("CLICKHOUSE_PORT", 9000)),
ValueError: invalid literal for int() with base 10: 'tcp://10.103.178.193:9000'
```
Somehow `'tcp://10.103.178.193:9000'` was passed into the `int()` function. I suspect it might have something to do with the environment variable `CLICKHOUSE_PORT`. Would you post the output of `kubectl exec -it <your-snuba-db-init-pod> -- env` here?

**Comment by prasadkris** [comment]
Thanks @junowong0114 , you are right, the `CLICKHOUSE_PORT` is set to `tcp://10.103.83.34:9000` as seen below, is there any workarounds we can use for fixing this? Thanks! 

```
env | grep CLICKHOUSE_PORT

CLICKHOUSE_PORT=tcp://10.103.83.34:9000
CLICKHOUSE_PORT_8123_TCP_PROTO=tcp
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_2888_TCP_PROTO=tcp
CLICKHOUSE_PORT_9009_TCP_PORT=9009
CLICKHOUSE_PORT_8123_TCP_ADDR=10.103.83.34
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_2888_TCP_PORT=2888
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_2181_TCP=tcp://10.105.94.126:2181
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_2181_TCP_PROTO=tcp
CLICKHOUSE_PORT_9000_TCP_ADDR=10.103.83.34
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_3888_TCP=tcp://10.105.94.126:3888
CLICKHOUSE_PORT_8123_TCP=tcp://10.103.83.34:8123
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_3888_TCP_ADDR=10.105.94.126
CLICKHOUSE_PORT_8123_TCP_PORT=8123
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_2181_TCP_PORT=2181
CLICKHOUSE_PORT_9009_TCP_PROTO=tcp
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_2888_TCP=tcp://10.105.94.126:2888
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT=tcp://10.105.94.126:2181
CLICKHOUSE_PORT_9009_TCP_ADDR=10.103.83.34
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_3888_TCP_PROTO=tcp
CLICKHOUSE_PORT_9000_TCP_PROTO=tcp
CLICKHOUSE_PORT_9009_TCP=tcp://10.103.83.34:9009
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_2888_TCP_ADDR=10.105.94.126
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_3888_TCP_PORT=3888
CLICKHOUSE_PORT_9000_TCP_PORT=9000
CLICKHOUSE_PORT_9000_TCP=tcp://10.103.83.34:9000
SENTRY_ZOOKEEPER_CLICKHOUSE_PORT_2181_TCP_ADDR=10.105.94.126
```

**Comment by junowong0114** [comment]
@prasadkris As you can see on the `snuba-db-init.job.yaml` ([link here](https://github.com/sentry-kubernetes/charts/blob/develop/sentry/templates/hooks/snuba-db-init.job.yaml#L76)), there are a few ways to inject env to `snuba-init` pod.
```
        env:
        - name: LOG_LEVEL
          value: debug
{{ include "sentry.snuba.env" . | indent 8 }}
{{- if .Values.snuba.dbInitJob.env }}
{{ toYaml .Values.snuba.dbInitJob.env | indent 8 }}
{{- end }}
        envFrom:
        - secretRef:
            name: {{ template "sentry.fullname" . }}-snuba-env
```
In `_helpers.tpl`, `sentry.snuba.env` is defined as below:
```
{{/*
Common Snuba environment variables
*/}}
{{- define "sentry.snuba.env" -}}
- name: SNUBA_SETTINGS
  value: /etc/snuba/settings.py
- name: DEFAULT_BROKERS
  value: {{ include "sentry.kafka.bootstrap_servers_string" . | quote }}
{{- end -}}
```
So you should check:
1. Have you added `CLICKHOUSE_PORT` to `_helpers.tpl`?
2. Have you added `CLICKHOUSE_PORT` to `.Values.snuba.dbInitJob.env`?
3. Have you added `CLICKHOUSE_PORT` to `sentry-snuba-env` secret?
4. Have you added `CLICKHOUSE_PORT` directly to `snuba-db-init.job.yaml`?

Remember, you can always run `helm install` with `--dry-run` flag to check the generated yaml files. You can redirect the results to a yaml file by `> dry-run.yaml` so that you can check the files in an IDE.

Please let us know which of the above is your case?

**Comment by prasadkris** [comment]
Thank you, @junowong0114 I am using the defalut settings and have not added `CLICKHOUSE_PORT` to any resources listed above. The only change from the default `values.yaml` is that I have disabled the internal clickhouse and have added `externalClickhouse:` with the external cluster details.

I tried adding the following line to the `sentry-snuba-env` secret and it did help to fix the error and I able to proceed further with the deployment, thank you! 

```
CLICKHOUSE_PORT: "OTAwMAo="
```
